pkg/proxy/proxy/transparent.go

```diff
@@ -98,28 +97,17 @@ type TransparentProxyService struct {
 }
 
 // NewTransparentProxyService creates a new TransparentProxyService.
-func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate, mode PolicyMode, policyFile string) TransparentProxyService {
+func NewTransparentProxyService(p *goproxy.ProxyHttpServer, ca *tls.Certificate, mode PolicyMode, pl *policy.Policy) TransparentProxyService {
```

Comment:
might be nice to make this a bit more defensive. if PolicyMode isn't Disabled, we should validate that the policy isn't nil. Alternatively, you could accept the policy by value instead of ptr. The former is probably sufficient.

Reply:
I was thinking that since the behaviour of enforcement mode is to have an allowlist rather than a blocklist, a nil policy should be treated as valid and just block everything.

Reply:
well it'll currently crash since it'll try to call Apply on nil so we need to fix it some way. i think i'd still lean towards doing the validation/defaulting here so if you'd like to keep the param nil, it'd need to create an empty instance.

Reply:
Oh good point. Added a check